src/main/java/maksym/fedorenko/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package maksym.fedorenko.bookstore.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import maksym.fedorenko.bookstore.dto.order.CreateOrderRequestDto;
+import maksym.fedorenko.bookstore.dto.order.OrderDto;
+import maksym.fedorenko.bookstore.dto.order.OrderItemDto;
+import maksym.fedorenko.bookstore.dto.order.UpdateOrderRequestDto;
+import maksym.fedorenko.bookstore.exception.CreateOrderException;
+import maksym.fedorenko.bookstore.exception.EntityNotFoundException;
+import maksym.fedorenko.bookstore.mapper.OrderItemMapper;
+import maksym.fedorenko.bookstore.mapper.OrderMapper;
+import maksym.fedorenko.bookstore.model.Order;
+import maksym.fedorenko.bookstore.model.OrderItem;
+import maksym.fedorenko.bookstore.model.ShoppingCart;
+import maksym.fedorenko.bookstore.repository.OrderItemRepository;
+import maksym.fedorenko.bookstore.repository.OrderRepository;
+import maksym.fedorenko.bookstore.repository.ShoppingCartRepository;
+import maksym.fedorenko.bookstore.service.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public OrderDto createOrder(Authentication authentication, CreateOrderRequestDto requestDto) {
+        ShoppingCart cart = shoppingCartRepository
+                .findByUserEmail(authentication.getName())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't create order because cart doesn't exist")
+                );
+        if (cart.getCartItems().size() == 0) {
+            throw new CreateOrderException("Can't create order from empty cart");
+        }
+        Order order = orderMapper.toOrder(cart);
+        order.setShippingAddress(requestDto.shippingAddress());
```

Comment:
you can pass the dto to omit the mapping here
```suggestion
        Order order = orderMapper.toOrder(cart, requestDto);
```